spec/requests/planning_application_list_spec.rb

```diff
@@ -32,68 +28,99 @@
+        expect(planning_application_json["started_at"]).to eq(json_time_format(planning_application.started_at))
+        expect(planning_application_json["determined_at"]).to eq(json_time_format(planning_application.determined_at))
+        expect(planning_application_json["created_at"]).to eq(json_time_format(planning_application.created_at))
+        expect(planning_application_json["invalidated_at"]).to eq(json_time_format(planning_application.invalidated_at))
+        expect(planning_application_json["withdrawn_at"]).to eq(json_time_format(planning_application.withdrawn_at))
+        expect(planning_application_json["ward"]).to eq(planning_application.ward)
+        expect(planning_application_json["awaiting_determination_at"]).to eq(json_time_format(planning_application.awaiting_determination_at))
+        expect(planning_application_json["in_assessment_at"]).to eq(json_time_format(planning_application.in_assessment_at))
+        expect(planning_application_json["awaiting_correction_at"]).to eq(json_time_format(planning_application.awaiting_correction_at))
+        expect(planning_application_json["agent_first_name"]).to eq(planning_application.agent_first_name)
+        expect(planning_application_json["agent_last_name"]).to eq(planning_application.agent_last_name)
+        expect(planning_application_json["agent_email"]).to eq(planning_application.agent_email)
+        expect(planning_application_json["applicant_first_name"]).to eq(planning_application.applicant_first_name)
+        expect(planning_application_json["applicant_last_name"]).to eq(planning_application.applicant_last_name)
+        expect(planning_application_json["applicant_email"]).to eq(planning_application.applicant_email)
+        expect(planning_application_json["applicant_phone"]).to eq(planning_application.applicant_phone)
+        expect(planning_application_json["site"]["address_1"]).to eq(planning_application.site.address_1)
+        expect(planning_application_json["site"]["address_2"]).to eq(planning_application.site.address_2)
+        expect(planning_application_json["site"]["town"]).to eq(planning_application.site.town)
+        expect(planning_application_json["site"]["county"]).to eq(planning_application.site.county)
+        expect(planning_application_json["site"]["postcode"]).to eq(planning_application.site.postcode)
+        expect(planning_application_json["site"]["uprn"]).to eq(planning_application.site.uprn)
+        expect(planning_application_json["questions"]).to eq(JSON.parse(planning_application.questions))
+        expect(planning_application_json["constraints"]).to eq(JSON.parse(planning_application.constraints))
+      end
+
+      context "for a granted planning application" do
+        let!(:planning_application) { create(:planning_application, :determined) }
+        let!(:decision) { create(:decision, :granted, user: reviewer, planning_application: planning_application) }
+
+        it "returns the accurate data" do
```

Comment:
I'm pondering whether these specs might be split into several different ones so we can more easily see where any failures are: for example, one spec for the site info, one for contact details of agent/applicant, or whatever? That way if we have a second failure after the test has stopped, it would be more quickly identified

Reply:
I definitely thought about that, but realised that unless I did a before(:all), which I try and avoid, it would be making dozens of setups and teardowns every test run.